main.nf

```diff
@@ -961,6 +979,80 @@ process krona {
 ================================================================================
 */
 
+// Co-assembly: prepare grouping for MEGAHIT and pool reads for SPAdes
+if (params.coassemble_group) {
+    // short reads
+    // group and set group as new name
+    ch_short_reads_assembly
+        .groupTuple(by: 1)
+        .map { names, grp, reads ->
+                if (!params.single_end) [ "group$grp", grp, reads.collect { it[0] }, reads.collect { it[1] } ]
+                else [ "group$grp", grp, reads.collect { it[0] }, [] ] }
+        .into { ch_short_reads_megahit; ch_grouped_short_reads }
+
+    // pool short reads for SPAdes assembly
+    process pool_short_reads {
+        tag "$name"
+
+        input:
+        set val(name), val(grp), file(reads1), file(reads2) from ch_grouped_short_reads
```

Comment:
`reads2` should be an empty list in that case, see https://github.com/skrakau/mag/blob/2c41ac0b6949819c9ed93d53fa9a2c6a1e3e8c72/main.nf#L990

But I have to admit, I didn't really test this (also because I would not be surprised, if it would cause problems already for previous versions). I will try to test.

Reply:
> reads2 should be an empty list in that case, see https://github.com/skrakau/mag/blob/2c41ac0b6949819c9ed93d53fa9a2c6a1e3e8c72/main.nf#L990

I see, great.

> But I have to admit, I didn't really test this (also because I would not be surprised, if it would cause problems already for previous versions). I will try to test.

A new test config could be added, such as test_single.config, with already available data but only forward reads?

Reply:
the tests using single end only fail for MetaBAT, also for previous releases of the pipeline. Maybe we can add this in future PRs, anyway, I also wanted to add tests using the new functionalities. However, the tests take quite a while already and maybe we can optimize something there...

Reply:
my bad, I probably forgot to set `min_length_unbinned_contigs = 1` and `max_unbinned_contigs = 2`...

Reply:
> the tests take quite a while already and maybe we can optimize something there...

the nf-core template suggests parallelization of CI tests by `strategy.matrix`. Haven't researched or tested that yet.